### PR TITLE
Update Regression Test to include gfsv17 prototypes that are after prototype 8, and turn on prognostic closure.

### PR DESCRIPTION
**Pull request by lisa-bengtsson**
# PR Checklist

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR
are specified below.

- [ ] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [ ] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Description

These PR’s in ufs-weather-model, fv3atm and ccpp/physics does the following:
Creates a set of _gfsv17 RT’s to reflect tests that are beyond prototype 8 targeted for GFSv17
Activates prognostic closure by setting namelist progsigma = true in *gfsv17 and pointing to new field_table in the RT’s
Addresses additional parametric tuning in progclosure_calc.F90
Adds a new gfsv17 field_table and diag_table 

The ufs-weather-model PR was also merged with a PR including outerloop for the wave model, and the ccpp/physics PR was merged with a fix for ccpp_prebuild. 

### Issue(s) addressed

https://github.com/ufs-community/ufs-weather-model/issues/1477

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel
- [x] cheyenne.gnu
- [x] gaea.intel
- [x] jet.intel
- [x] wcoss2.intel
- acorn.intel (skip for queue upgrade testing)
- [ ] opnReqTest for newly added/changed feature
- [ ] CI

## Dependencies

https://github.com/ufs-community/ccpp-physics/pull/18
https://github.com/NOAA-EMC/fv3atm/pull/598

### CODE REVIEW

**Comment by lisa-bengtsson** [comment]
@ChunxiZhang-NOAA @junwang-noaa @yangfanglin @DeniseWorthen @JessicaMeixner-NOAA can you take a close look at these tests and provide feedback. I'm working on the CCPP PR to go with this PR as well, it should be merged together. New baselines would be needed for the new tests. Thank you!

File: tests/tests/control_gfsv17 [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+export KNOB_UGWP_NSLOPE=1
+export DO_UGWP_V0=.true.
+export DO_GSL_DRAG_LS_BL=.false.
+export DO_GSL_DRAG_SS=.true.
+export DO_GSL_DRAG_TOFD=.false.
+export DO_UGWP_V1_OROG_ONLY=.false.
+export DO_UGWP_V0_NST_ONLY=.false.
+export LDIAG_UGWP=.false.
+
+# P7 CA
```
Comment (JessicaMeixner-NOAA) [inline tests/tests/control_gfsv17]: Might be time to remove "P*" labels as we start fresh for gfsv17?

Comment (lisa-bengtsson) [reply tests/tests/control_gfsv17]: Yes, I can remove those.

**Comment by lisa-bengtsson** [comment]
I added export DNATS=0 in the new gfsv17 tests in order to not exclude any tracers from advection since the default value is DNATS=2, I wonder if this (DNATS = 0) should also be the case in the test cpld_control_noaero_p8?

**Comment by ChunxiZhang-NOAA** [comment]
> cpld_control_noaero_p8

Good catch. DNATS should be 0 for cpld_control_noaero_p8, otherwise TKE will not be advected.

**Comment by junwang-noaa** [comment]
@lisa-bengtsson @ChunxiZhang-NOAA @JessicaMeixner-NOAA SInce you are adding the GFSv17 tests, would you please remove the control related tests from GFSv16? Otherwise the RT is taking too long to finish for code commit.

**Comment by lisa-bengtsson** [comment]
> > cpld_control_noaero_p8
> 
> Good catch. DNATS should be 0 for cpld_control_noaero_p8, otherwise TKE will not be advected.

Ok, I will update those noaero tests.

**Comment by lisa-bengtsson** [comment]
> @lisa-bengtsson @ChunxiZhang-NOAA @JessicaMeixner-NOAA SInce you are adding the GFSv17 tests, would you please remove the control related tests from GFSv16? Otherwise the RT is taking too long to finish for code commit.

@junwang-noaa can you list which control tests can be removed? So far I'm only adding cpld_control ORT tests, and one uncoupled control_gfsv17. Thanks.

**Comment by junwang-noaa** [comment]
This is good question for GFSv17 test owners. My understanding is that all the control related test except control_p8 related tests

**Comment by lisa-bengtsson** [comment]
@junwang-noaa Who are the code owners for the GFSv16 tests? Perhaps what should be done is to remove the GFSv16 tests, and create ORT tests based on control_gfsv17 instead? Although this doesn't help in removing any tests... It is not clear to me what can be removed. Thanks.

**Comment by junwang-noaa** [comment]
@lisa-bengtsson We can't run GFSv16 configuration with the control related test any more. Yes, we need to remove all these tests and create ORT tests based on control_gfsv17. Any new feature gfsv17 test needs to be based on control_gfsv17.

**Comment by lisa-bengtsson** [comment]
> @lisa-bengtsson We can't run GFSv16 configuration with the control related test any more. Yes, we need to remove all these tests and create ORT tests based on control_gfsv17. Any new feature gfsv17 test needs to be based on control_gfsv17.

@junwang-noaa Note that replacing the previous gfsv16 tests with gfsv17 tests will not reduce the total tests from this PR as we are adding additional cpld_...gfsv17 tests and not removing existing cpld...p8 tests. This is because it was suggested that those should be replaced by cpld_...gefsv13 tests in the future.  Your suggestion sounds like it is something that the test owners and code managers need to do, I would propose in a separate PR.

**Comment by DeniseWorthen** [comment]
@lisa-bengtsson You've added new nml.IN files for gfsv17. Other than the ``progsima`` variable, these are identical to the  existing p8 versions. Do you need to add the new nml files? Or, do you envision that the future gefsv13 (currently p8) would eventually diverge too much from what is the gfsv17 for that to make sense.

Also, I believe you will also need to add a bmark_restart_gfsv17 test. 

Finally, there's a problem w/ your .gitmodules, which points to a non-existent FV3 fork (lisa.bengtsson).

**Comment by lisa-bengtsson** [comment]
> @lisa-bengtsson You've added new nml.IN files for gfsv17. Other than the `progsima` variable, these are identical to the existing p8 versions. Do you need to add the new nml files? Or, do you envision that the future gefsv13 (currently p8) would eventually diverge too much from what is the gfsv17 for that to make sense.
> 
> Also, I believe you will also need to add a bmark_restart_gfsv17 test.
> 
> Finally, there's a problem w/ your .gitmodules, which points to a non-existent FV3 fork (lisa.bengtsson).

Hi Denise, thanks I updated the .gitmodules. 

My understanding is that the final code base and configuration of gefsv13 is not decided yet, but will have to be finalized within the next 2 months. gfsv17 will continue to be developed until the early summer 2023, so in the end they will diverge. @yangfanglin can you and/or Avichal weigh in on the strategy for these tests?

**Comment by JessicaMeixner-NOAA** [comment]
@lisa-bengtsson @jkbk2004 would it make sense for me to make a PR to this PR with the updates from #1502 ?

**Comment by lisa-bengtsson** [comment]
@JessicaMeixner-NOAA that is OK with me, it looks like I'm still scheduled to merge tomorrow.

**Comment by jkbk2004** [comment]
@lisa-bengtsson @JessicaMeixner-NOAA Please, go ahead to combine #1502. In terms of the commit date, Dec 01 is most likely. AQM v7 release PR is coming in tomorrow. So it will be possible to merge in this pr on Dec 02 morning. Would that be ok?

**Comment by lisa-bengtsson** [comment]
Yes, that works for me. Thanks.

**Comment by jkbk2004** [comment]
@lisa-bengtsson @JessicaMeixner-NOAA We are currently working on #1505 (ready to merge today). #1511 is a hot-fix for HR1 before this pr. So, we need to ask to delay a day more for this pr. I think we can start testing this pr tomorrow. Merging Monday morning is possible. Please, let me know if this schedule change works.

**Comment by JessicaMeixner-NOAA** [comment]
Thanks for the update @jkbk2004 !

**Comment by jkbk2004** [comment]
@lisa-bengtsson #1511 was merged. We can start working on this pr today. Can you sync up branch and resolve the conflicts?

**Comment by lisa-bengtsson** [comment]
@jkbk2004 I brought the PR up to date. @junwang-noaa @DeniseWorthen can you look at rt.conf to see if this is in agreement with our discussion? I removed the control tests that had an equivalent control_p8 test.

**Comment by DeniseWorthen** [comment]
@lisa-bengtsson What is the updated input data required for this PR? Is it locally staged somewhere on Hera?

**Comment by lisa-bengtsson** [comment]
@DeniseWorthen there's no updated input data as far as I'm aware? just updated source code in ccpp/physics and some updated tests. Maybe I'm misunderstanding what you are asking?

**Comment by DeniseWorthen** [comment]
@lisa-bengtsson The PR template has a checkmark on the item that new input data is required. I didn't think that @JessicaMeixner-NOAA was adding a new mod_def for the outer loop test so I didn't know what the new input data would be.

**Comment by lisa-bengtsson** [comment]
@DeniseWorthen ah, I think I had confused that with "new baselines", sorry I unchecked it. @JessicaMeixner-NOAA does your update require new data?

**Comment by JessicaMeixner-NOAA** [comment]
No new input data on my side either.

File: tests/parm/nems.configure.cpld_outwav.IN [hunk 2]
```diff
@@ -0,0 +1,141 @@
+#############################################
```
Comment (DeniseWorthen) [inline tests/parm/nems.configure.cpld_outwav.IN]: I don't see any test using this nems.configure file. Is there are reason it is needed?

File: tests/parm/nems.configure.cpld_noaero_outwav.IN [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+     MED med_phases_prep_wav_accum
+   @
+   OCN -> MED :remapMethod=redist
+   WAV -> MED :remapMethod=redist
+   MED med_phases_post_ocn
+   MED med_phases_post_wav
+   MED med_phases_restart_write
+@
+::
+::
```
Comment (DeniseWorthen) [inline tests/parm/nems.configure.cpld_noaero_outwav.IN]: Remove the second "::"

File: tests/parm/nems.configure.cpld_outwav.IN (hunk 2 above)
Comment (JessicaMeixner-NOAA) [reply tests/parm/nems.configure.cpld_outwav.IN]: We're just adding the one test for now, but it's very likely that we'll use this configuration in EP4, so the other coupled cases, so I was being proactive and including it now.  It can be removed for the moment.

Comment (DeniseWorthen) [reply tests/parm/nems.configure.cpld_outwav.IN]: So why is the added test a 'noaero' test and not just an outer-loop version of cpld_control_gfsv17?

Comment (JessicaMeixner-NOAA) [reply tests/parm/nems.configure.cpld_outwav.IN]: Because for now we wanted to keep the testing burden low and we know for sure we will use this for GFSv17, when we know for sure this will also be for GEFS, we can update the p8 tests as well.

**Comment by jkbk2004** [comment]
I will add new BL_DATE.

**Comment by lisa-bengtsson** [comment]
> I will add new BL_DATE.

Thank you, I don't know how you usually do it, but in this case only the new test tests/tests/cpld_control_gfsv17 needs new baselines. The update should not change the results of any other tests.

**Comment by jkbk2004** [comment]
Yeah, I will manually create the one and copy over all others.

**Comment by lisa-bengtsson** [comment]
> Yeah, I will manually create the one and copy over all others.

I was mistaken @jkbk2004 the tests that have updates in the DNATS variable needs new baselines as well, those are the ones with noareo_p8 in the name. The reason for the baseline change is that they erroneously had DNATS=2 before.

**Comment by BrianCurtis-NOAA** [comment]
@lisa-bengtsson Are you intending to remove tests from rt.conf and tests/tests. I also noticed that control_debug is being deleted but not removed from rt.conf.

Just making sure this is intended and to correct the issue with control_debug.

**Comment by lisa-bengtsson** [comment]
> @lisa-bengtsson Are you intending to remove tests from rt.conf and tests/tests. I also noticed that control_debug is being deleted but not removed from rt.conf.
> 
> Just making sure this is intended and to correct the issue with control_debug.

@BrianCurtis-NOAA yes, I discussed with @junwang-noaa offline who suggested to remove control gfsv16 tests that had equivalent control_p8 tests. I'm hoping she can confirm that these changes are correct. The control_debug tests should also be removed as there is a control_p8_debug test. I corrected that in rt.conf.

**Comment by BrianCurtis-NOAA** [comment]
@lisa-bengtsson Thanks! I just found out about the discussion to remove RT's from @JessicaMeixner-NOAA as well. I'm very happy to see tests disappear!

**Comment by lisa-bengtsson** [comment]
@BrianCurtis-NOAA yes, I agree with that!

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: gnu
Job: RT
[RT] Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/1103466935/20221203171511/ufs-weather-model
Please make changes and add the following label back: hera-gnu-RT

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: intel
Job: RT
[RT] Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/1103466935/20221203170009/ufs-weather-model
[RT] Error: Test cpld_control_noaero_p8 013 failed in check_result failed 
[RT] Error: Test cpld_control_noaero_p8 013 failed in run_test failed 
[RT] Error: Test cpld_debug_noaero_p8 016 failed in check_result failed 
[RT] Error: Test cpld_debug_noaero_p8 016 failed in run_test failed 
Please make changes and add the following label back: hera-intel-RT

**Comment by lisa-bengtsson** [comment]
@BrianCurtis-NOAA sorry I was mistaken these tests also needs new baselines:
Test cpld_control_noaero_p8
Test cpld_control_noaero_p8 
Test cpld_debug_noaero_p8 
Test cpld_debug_noaero_p8 

The reason is that these updates correct a bug of setting DNATS=2 in case of no aerosol, but doing so we do not advect the last two fields defined in field_table. The correct setting should be DNATS=0 if aerosols are turned off. That is why results change and new baselines are needed also for these tests.

Thanks

**Comment by jkbk2004** [comment]
@lisa-bengtsson I am deleting control/control_restart in rt_gnu.conf.

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by JessicaMeixner-NOAA — APPROVED** [review]

**Comment by jkbk2004** [comment]
We will skip jenkins-ci. We can start merging process.

**Comment by ChunxiZhang-NOAA** [comment]
@jkbk2004 ccpp-physics PR was merged.

**Comment by jkbk2004** [comment]
> @jkbk2004 ccpp-physics PR was merged.

thanks!

File: .gitmodules [hunk 4]
```diff
@@ -1,7 +1,9 @@
 [submodule "FV3"]
   path = FV3
-  url = https://github.com/NOAA-EMC/fv3atm
-  branch = develop
+#  url = https://github.com/NOAA-EMC/fv3atm
+#  branch = develop
+ url = https://github.com/lisa-bengtsson/fv3atm
+ branch = progc_update
```
Comment (BrianCurtis-NOAA) [inline .gitmodules]: @lisa-bengtsson Please revert these changes.

**Review by BrianCurtis-NOAA — CHANGES_REQUESTED** [review]

**Comment by BrianCurtis-NOAA** [comment]
FV3 hash is https://github.com/NOAA-EMC/fv3atm/commit/92b0386f59163005f6cc629847a60ac0ba77201e

**Comment by BrianCurtis-NOAA** [comment]
@lisa-bengtsson FV3 hash isn't updated yet. Should be 92b0386, but is currently bd39951

**Comment by lisa-bengtsson** [comment]
> @lisa-bengtsson FV3 hash isn't updated yet. Should be 92b0386, but is currently bd39951

ok, let me try again.

**Comment by lisa-bengtsson** [comment]
> > @lisa-bengtsson FV3 hash isn't updated yet. Should be 92b0386, but is currently bd39951
> 
> ok, let me try again.

hm, I'm confused, should I not to "git pull upstream develop"? It doesn't seem to point to the correct one if I do that?

**Comment by BrianCurtis-NOAA** [comment]
> > > @lisa-bengtsson FV3 hash isn't updated yet. Should be 92b0386, but is currently bd39951
> > 
> > 
> > ok, let me try again.
> 
> hm, I'm confused, should I not to "git pull upstream develop"? It doesn't seem to point to the correct one if I do that?

You should do a `git remote update` then `git checkout upstream/develop` then it should be on the correct hash. You can check with `git submodule status`, each submodules hash will show.

**Comment by BrianCurtis-NOAA** [comment]
I gave one wrong instruction. in FV3 you would do the `git remote update` then `git checkout upstream/develop` at that point it should be the hash you're looking for, then in UFSWM you can do a `git submodule status` to see the FV3 hash, then `git add FV3` will get it into UFSWM.

**Comment by BrianCurtis-NOAA** [comment]
Looks good now. Thanks!

**Comment by lisa-bengtsson** [comment]
@BrianCurtis-NOAA Ok, thank you for your help!

**Review by BrianCurtis-NOAA — APPROVED** [review]
